src/jbms/argparse.cpp

```diff
@@ -5,11 +5,12 @@
 #include <cctype>
 #include <iostream>
 
-// Needed to determine terminal width
-// TODO: add support for Windows
-#include <sys/ioctl.h>
-#include <stdio.h>
-#include <unistd.h>
+#if defined(_WIN64) || (defined(__CYGWIN__) && !defined(_WIN32))
```

Comment:
As mentioned, I don't think you need to check for `_WIN64`.

Also, does the `ioctl` not work on cygwin?  I would suppose that on cygwin you might be using something other than the normal command prompt and then would `GetConsoleScreenBufferInfo` still work?

Reply:
I'm really sorry 😅 this is my mistake. I didn't understand what you were saying previously actually it was late at night when I was working on this I was texting one of my friend, waiting for someone's reply and working on this as well 😅 because of that I thought you said remove `_WIN32` and honestly I didn't even bother to check how reasonable your point is. Now I'm going to fix my mistakes.

Reply:
> Also, does the ioctl not work on cygwin? I would suppose that on cygwin you might be using something other than the 
> normal command prompt and then would GetConsoleScreenBufferInfo still work?

The `WIN32API` can successfully be called from `cygwin` but it implies that you must have `win32` packages installed.
`GetConsoleScreenBufferInfo` will work on cygwin systems.

Reply:
`sys/ioctl.h` there is no such file on cygwin.